internal/api/cluster_installation.go

```diff
@@ -381,11 +381,10 @@ func handleMigrateClusterInstallations(c *Context, w http.ResponseWriter, r *htt
 		"target-cluster-id": mcir.TargetClusterID,
 	})
 
-	IsMigrated := false
 	filter := &model.ClusterInstallationFilter{
 		ClusterID:      mcir.SourceClusterID,
 		InstallationID: mcir.InstallationID,
-		IsMigrated:     &IsMigrated,
+		TargetCluster:  mcir.TargetClusterID,
```

Comment:
Sorry I might have not been clear enough.

So I was thinking more of just doing a second query to the database and filtering in code like so:
```go
// This whole logic could be a separate method - `getClusterInstallationsForMigration` or something. 
toMigrateFilter := &model.ClusterInstallationFilter{
		ClusterID:      mcir.SourceClusterID,
		InstallationID: mcir.InstallationID,
                 Paging:         model.AllPagesNotDeleted(),
}
alredyMigratedFilter := &model.ClusterInstallationFilter{
		ClusterID:      mcir.TargetClusterID,
		InstallationID: mcir.InstallationID,
                 Paging:         model.AllPagesNotDeleted(),
}

// Here query both
sourceClusterCIs, err := c.Store.GetClusterInstallations(toMigrateFilter)
...
migratedCIs, err := c.Store.GetClusterInstallations(alredyMigratedFilter)
...

toMigrate := []*model.ClusterInstallation{}
for _, ci := range sourceClusterCIs {
	migrate := true
	for _, migrated := range migratedCIs {
		if ci.InstallationID == migrated.InstallationID {
			migrate = false
			break
		} 
	}
	if migrate {
		toMigrate = append(toMigrate, ci)
	}
}
```

Filtering on SQL level makes sense also, but I would suggest introducing a separate method for that like `GetClusterInstallationsForMigration` to not leak migration logic into a more generic construct that is `ClusterInstallationFilter`. With some thoughtful refactoring, we could probably reuse some parts of `getClusterInstallations` there.

Reply:
All good. I thought about the above loop based comparison but felt it would be more efficient if we can handle it at SQL level. (Performance wise)
I'll check if introducing seperate method like `GetClusterInstallationsForMigration` is not too much impact on existing logic otherwise it will be simpler to go with your suggestion.